### ISSUE TITLE
Copilot Chat doesn't retain history of chat message when VS Code has to reload.

### ISSUE THREAD

**Issue by webreidi** [microsoft/vscode-copilot-release#2078]
- Copilot Chat Extension Version: v0.22.2024102805 (pre-release)
- VS Code Version: 1.95.0-insiders
- OS Version: Windows 11
- Logs:

Steps to Reproduce:

1. As Copilot chat for help in creating a .NET API
2. Follow the step to create a new .NET Application.
3. Window reloads during the process of creating a new application and the steps listed by Copilot are no longer available.

Copilot chat should retain the history between reloads.

**Comment by joyceerhl** [comment]
Have you tried using the `@workspace /new` command in chat to create new projects?

**Comment by webreidi** [comment]
Yes, I tried that, but it doesn't properly create a .NET project.

**Comment by roblourens** [comment]
> Window reloads during the process of creating a new application and the steps listed by Copilot are no longer available.

So you don't see the previous chat session in the chat view? This still works for me. Can you

- share a screenshot to make sure I understand
- Check for errors in the Window output channel?

**Comment by webreidi** [comment]
I have VS Code with nothing and ask copilot to help me create a new .NET API.
![Image](https://github.com/user-attachments/assets/9552fdc2-95d7-49d0-b3ce-4ebe8ae579a6)

After following step 1 to create the new .NET project and opening it to do step 2, the chat is empty.

![Image](https://github.com/user-attachments/assets/354e2d6a-4822-4a8f-a302-8e983c880ab1)

**Comment by roblourens** [comment]
I see, thanks. If you use `@workspace /new`, then it will actually create a new workspace with files for you, and if you follow that flow, it will transfer your chat session into the new workspace.

The only other way to move a chat session from one workspace to another is to use the "Export Chat" command and then "Import Chat" in the new window. This is a bit finnicky, but we think it makes sense in general for chats to be workspace-specific.

**Comment by roblourens** [comment]
Feature request for that here though: https://github.com/microsoft/vscode-copilot-release/issues/991

**Comment by webreidi** [comment]
Unfortunately, I couldn't get @workspace to create a project that used modern .NET standards, whereas a regular chat session allowed for a dialogue that allowed it to build and modify to get a solution that worked. I've watched several customers use Claude instead of Copilot to do this very task over the past couple of weeks because Copilot is not meeting their needs.
